### PR TITLE
chore: drop kitkat support

### PR DESCRIPTION
**Pull request by demetris-manikas**
A lot of cleanup. All tests pass. Please test on iOS.

### CODE REVIEW

**Comment by brodybits** [comment]
* It looks like AES support is dropped. Can you confirm? If so, doesn't this affect the other platforms? Shouldn't the documentation be updated?
* I think this change needs a new major version number due to both dropping old Android version and apparently dropping support for an encryption algorithm (AES). Or am I missing something here?

File: src/android/SecureStorage.java [hunk 1]
```diff
@@ -21,10 +21,9 @@
 public class SecureStorage extends CordovaPlugin {
     private static final String TAG = "SecureStorage";
 
-    private static final boolean SUPPORTS_NATIVE_AES = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
-    private static final boolean SUPPORTED = Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT;
+    private static final boolean SUPPORTED = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
 
-    private static final String MSG_NOT_SUPPORTED = "API 19 (Android 4.4 KitKat) is required. This device is running API " + Build.VERSION.SDK_INT;
+    private static final String MSG_NOT_SUPPORTED = "API 26 (Android 8.0.0 Oreo) is required. This device is running API " + Build.VERSION.SDK_INT;
```
[the remark this answers is not in the document]
Comment (demetris-manikas) [reply src/android/SecureStorage.java]: Sure.

**Comment by ggozad** [comment]
> * It looks like AES support is dropped. Can you confirm? If so, doesn't this affect the other platforms? Shouldn't the documentation be updated?

I haven't had the time to look into this properly but from a quick glance it seems to me that native AES is used for Android now. For iOS this is unnecessary as the native keychain is used. So AES is still used for Android but we do not rely anymore on sjcl for it.

> * I think this change needs a new major version number due to both dropping old Android version and apparently dropping support for an encryption algorithm (AES). Or am I missing something here?

SecureStorage will be bumped to a major version yes. The current version can be used by folks needing compatibility with older version of android.

**Comment by demetris-manikas** [comment]
No. AES is not dropped. Usage of sjcl AES  that supported KitKat is dropped.
Native AES is used for LOLLIPOP and up.
 the docs will be updated to mention the drop of support for KitKat.
new major version will be assigned.

**Comment by demetris-manikas** [comment]
Can you guys give a minute?

**Comment by ggozad** [comment]
Thank you very much Demetris, this looks good. I should be able to review and test on Android as well as iOS next week. 
Could you please update the docs.

Is there someone who can verify tests pass on Windows?

**Comment by demetris-manikas** [comment]
@brodybits Thanks for your comments.

**Comment by brodybits** [comment]
Thakns @demetris-manikas for the updates and clarification.

I would be happy to give it a try on Windows, probably sometime next week. Please ping me here in case I forget.

**Comment by ggozad** [comment]
I am good with iOS and the code review. If @brodybits can verify all ok under windows I will merge and release.

**Comment by demetris-manikas** [comment]
@brodybits Hi. Can you please perform the tests on windows so that we can go on with this? 
Thanks in advance.

**Comment by brodybits** [comment]
I will need a few more days, apologies.

**Pull request merged**